prereise/gather/hiflddata/data_trans.py

```diff
@@ -0,0 +1,951 @@
+        "SHELBY",
+        "SAN AUGUSTINE",
+        "SABINE",
+        "JASPER",
+        "NEWTON",
+        "ORANGE",
+        "JEFFERSON",
+        "LIBERTY",
+        "HARDIN",
+        "TYLER",
+        "POLK",
+        "TRINITY",
+        "WALKER",
+        "SAN JACINTO",
+        "DALLAM",
+        "SHERMAN",
+        "HANSFORD",
+        "OCHLTREE" "LIPSCOMB",
+        "HARTLEY",
+        "MOORE",
+        "HUTCHINSON",
+        "HEMPHILL",
+        "RANDALL",
+        "DONLEY",
+        "PARMER",
+        "BAILEY",
+        "LAMB",
+        "HALE",
+        "COCHRAN",
+        "HOCKLEY",
+        "LUBBOCK",
+        "YOAKUM",
+        "TERRY",
+        "LYNN",
+        "GAINES",
+    ]
+
+    sd_west = ["LAWRENCE", "BUTTE", "FALL RIVER"]
+    # nm_east = ['CURRY', 'LEA', 'QUAY', 'ROOSEVELT', 'UNION']
+    mt_east = [
+        "CARTER",
+        "CUSTER",
+        "ROSEBUD",
+        "PRAIRIE",
+        "POWDER RIVER",
+        "DANIELS",
+        "MCCONE",
+        "DAWSON",
+        "RICHLAND",
+        "FALLON",
+        "GARFIELD",
+        "ROOSEVELT",
+        "PHILLIPS",
+        "SHERIDAN",
+        "VALLEY",
+        "WIBAUX",
+    ]
+    sub = open("output/sub.csv", "w", newline="")
+    csv_writer = csv.writer(sub)
+    csv_writer.writerow(
+        [
+            "sub_id",
+            "name",
+            "zip",
+            "lat",
+            "lon",
+            "interconnect",
+            "zone_id",
+            "type",
+            "state",
+        ]
+    )
+    sub_code = {}
+    re_code = {}
+    for index, row in clean_data.iterrows():
+        if row["STATE"] in West:
+            re = "Western"
+        elif row["STATE"] in Uncertain:
+            if row["STATE"] == "TX":
+                if row["COUNTY"] in tx_west:
+                    re = "Western"
+                elif row["COUNTY"] in tx_east:
+                    re = "Eastern"
+                else:
+                    re = "Texas"
+
+            elif row["STATE"] == "SD":
+                if row["COUNTY"] in sd_west:
+                    re = "Western"
+                else:
+                    re = "Eastern"
+            elif row["STATE"] == "MT":
+                if row["COUNTY"] in mt_east:
+                    re = "Eastern"
+                else:
+                    # code = zone_dic1[(row['STATE'],re)]
+                    re = "Western"
+        else:
+            re = "Eastern"
+
+        code = zone_dic1[(row["STATE"], re)]
+        sub_code[row["ID"]] = code
+        re_code[row["ID"]] = re
+        csv_writer.writerow(
+            [
+                row["ID"],
+                row["NAME"],
+                row["ZIP"],
+                row["LATITUDE"],
+                row["LONGITUDE"],
+                re,
+                code,
+                row["TYPE"],
+                row["STATE"],
+            ]
+        )
+
+    sub.close()
+
+    return re_code, sub_code
+
+
+def write_bus(clean_data, sub_code, re_code, kv_dict):
```

Comment:
Can we separate the data filtering (building `missing_sub`) from the file writing (for which we can use the `to_csv` data frame methods)?